tactics/hints.ml

```diff
@@ -919,7 +905,7 @@ let fresh_global_or_constr env sigma poly cr =
     else begin
       if isgr then
         warn_polymorphic_hint (pr_hint_term env sigma ctx cr);
-      Lib.add_anonymous_leaf (input_context_set ctx);
+      Declare.declare_universe_context false ctx;
```

Comment:
I know this problem was already here before but that looks completely wrong to me. Global modification of the state should be the role of the toplevel caller of this code. I am not asking you to fix it in this PR, but it should definitely be fixed at some point.